readthedocs/oauth/services/github.py

```diff
@@ -325,31 +324,20 @@ def setup_webhook(self, project, integration=None):
                 recv_data = resp.json()
                 integration.provider_data = recv_data
                 integration.save()
-                log.info(
-                    'GitHub webhook creation successful for project.',
-                    project_slug=project.slug,
-                )
+                log.info('GitHub webhook creation successful for project.')
                 return (True, resp)
 
             if resp.status_code in [401, 403, 404]:
-                log.info(
-                    'GitHub project does not exist or user does not have permissions.',
-                    project_slug=project.slug,
-                )
-
-            # All other status codes will flow to the `else` clause below
+                log.info('GitHub project does not exist or user does not have permissions.')
 
         # Catch exceptions with request or deserializing JSON
         except (RequestException, ValueError):
-            log.exception(
-                'GitHub webhook creation failed for project.',
-                project_slug=project.slug,
-            )
+            log.exception('GitHub webhook creation failed for project.')
+        # TODO: I think this "else" clause here is executed when the `try` did
+        # not raise an exception. So we should probably delete it since it's
+        # calling log.error when there was no error.
         else:
-            log.error(
-                'GitHub webhook creation failed for project.',
-                project_slug=project.slug,
-            )
+            log.error('GitHub webhook creation failed for project.')
```

Comment:
@stsewd does this TODO note sound accurate to you? Can we remove the `else` from here?

Reply:
Example on GitLab that has the same logic: https://sentry.io/organizations/read-the-docs/issues/2827543888/?project=148442&query=is%3Aunresolved&statsPeriod=14d

Take a look at the reason: `404 - Not Found`.